### PR TITLE
samples: add missing sample.yaml files

### LINKED ISSUE #27813: samples without sample.yaml
Parallel to #27805 these are samples that lack a `sample.yaml` so may not be participating in CI as expected.

* [ ] samples/boards/intel_s1000_crb/dmic
* [ ] samples/subsys/mgmt/osdp

~* [ ] samples/testing/integration (has a `testcase.yaml` instead)~ apparently this is not a sample

### PR DESCRIPTION
**Pull request by jenmwms**
Some samples were without sample.yaml. This adds
them.

Fixes #27813

Signed-off-by: Jennifer Williams <jennifer.m.williams@intel.com>

### CODE REVIEW

**Comment by jenmwms** [comment]
From what I understand to be happening, the build failures are not necessarily introduced by this patch, but rather made visible in CI by adding the sample.yaml. The patch implements missing sample.yaml, which is correctly found by sanitycheck. So, unless I'm mistaken, I should submit a bug for these?
- `iotdk` for `samples/subsys/mgmt/osdp/`
- `intel_s1000_crb` for `samples/boards/intel_s1000_crb/dmic/sample.board.intel_s1000_crb.dmic`

**Comment by nashif** [comment]
>     * `intel_s1000_crb` for `samples/boards/intel_s1000_crb/dmic/sample.board.intel_s1000_crb.dmic`

this https://github.com/zephyrproject-rtos/zephyr/pull/27946 should fix it.

**Comment by nashif** [comment]
>     * `iotdk` for `samples/subsys/mgmt/osdp/`

this just needs a filter, has nothing to do with iotdk. 

filter: dt_compat_enabled_with_alias("gpio-leds", "led0") and CONFIG_SERIAL should do it, but OSDP needs a fix as well, you can add it here as another commit:

```
diff --git a/subsys/mgmt/osdp/Kconfig b/subsys/mgmt/osdp/Kconfig
index 1b3aa3e1b8..a5b4de60c4 100644
--- a/subsys/mgmt/osdp/Kconfig
+++ b/subsys/mgmt/osdp/Kconfig
@@ -8,11 +8,8 @@ menuconfig OSDP
        bool "Open Supervised Device Protocol (OSDP) driver"
        select RING_BUFFER
        select SERIAL
-       select SERIAL_HAS_DRIVER
        select SERIAL_SUPPORT_INTERRUPT
-       select UART_INTERRUPT_DRIVEN
        select ENTROPY_GENERATOR
-       select TEST_RANDOM_GENERATOR
        help
          Add support for Open Supervised Device Protocol (OSDP)
```

**Comment by jenmwms** [comment]
I incorporated the suggestions to `osdp/sample.yaml` from https://github.com/zephyrproject-rtos/zephyr/commit/de6500d7d9c6f87b88d5f803504deacc6a70cf69 in https://github.com/zephyrproject-rtos/zephyr/pull/27949

**Comment by pabigot** [comment]
Well, this certainly is finding a lot of problems in the samples....  This looks like it'll be good, but I'll wait until tests pass before reviewing for real.

**Comment by jenmwms** [comment]
I applied the filter (thanks for the tip!) and rebased now that the #27949 merged.

**Comment by jenmwms** [comment]
> This looks like it'll be good, but I'll wait until tests pass before reviewing for real.

@pabigot Thanks for the review, glad this patch is helpful! Please take another look.

File: samples/subsys/mgmt/osdp/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  name: osdp
+tests:
+  sample.subsys.mgmt.osdp:
+    tags: osdp
+    filter: dt_compat_enabled_with_alias("gpio-leds", "led0") and CONFIG_SERIAL
+    harness: osdp
+    integration_platforms:
+      - frdm_k64f
+
+# FIXME: exclude due to build error
```
Comment (nashif) [inline samples/subsys/mgmt/osdp/sample.yaml:12]: there is not reason for this to stay

**Review by nashif — CHANGES_REQUESTED** [review]

File: samples/subsys/mgmt/osdp/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+sample:
+  description: OSDP Sample
+  name: osdp
+tests:
+  sample.subsys.mgmt.osdp:
```
Comment (nashif) [inline samples/subsys/mgmt/osdp/sample.yaml:5]: remove subsys

File: samples/subsys/mgmt/osdp/sample.yaml (hunk 1 above)
Comment (jenmwms) [reply samples/subsys/mgmt/osdp/sample.yaml]: Thanks! removed it.

File: samples/subsys/mgmt/osdp/sample.yaml (hunk 2 above)
Comment (jenmwms) [reply samples/subsys/mgmt/osdp/sample.yaml]: Thanks! The naming scheme seems to vary across samples. Removed this here :)

**Review by nashif — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Pull request merged**